src/app/shared/services/sso-auth.service.ts

```diff
@@ -16,13 +21,25 @@ export class SSOAuthService extends AuthService {
         return Cookie.get('auth_token');
     }
 
+    public getCurrentUser(): Observable<User> {
+        if (!this.user) {
+            return this.api.whoAmI().pipe(
+                tap((user: User) => this.user = user)
+            );
+        } else {
+            return of(this.user);
+        }
+    }
+
     public login(username: string, password: string) {
         return this.api.getAuthToken(username, password).pipe(
-            map(res => Cookie.set('auth_token', res.token))
+            map(res => Cookie.set('auth_token', res.token)),
```

Comment:
and not here :-)

Reply:
could use tap here, but it doesn't make any difference because the result isn't being piped into the next function.

Reply:
I'm not sure what login should return (but it has to return an observable).